packages/salesforcedx-utils/test/jest/cli/cliCommandExecution.test.ts

```diff
@@ -0,0 +1,203 @@
+import { CancellationToken, CliCommandExecution, Command } from '../../../src';
+import {
+  CANCELATION_INTERVAL,
+  KILL_CODE,
+  NO_PID_ERROR,
+  NO_STDERR_ERROR,
+  NO_STDOUT_ERROR
+} from '../../../src/cli/cliCommandExecution';
+import { Observable } from 'rxjs/Observable';
+import * as kill from 'tree-kill';
+jest.mock('tree-kill');
+
+const treeKillMocked = jest.mocked(kill);
+
+describe('CliCommandExecution Unit Tests.', () => {
+  const testCommand: Command = {
+    command: 'do a thing',
+    args: ['arg1', 'arg2'],
+    toCommand: jest.fn()
+  };
+  let testChildProcess: any;
+  let testCancelationToken: CancellationToken;
+  let fromEventSpy: jest.SpyInstance;
+  let intervalSpy: jest.SpyInstance;
+  let subscribeSpy: jest.SpyInstance;
+  let unsubscribeSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    testChildProcess = {
+      pid: 1234,
+      stdout: jest.fn(),
+      stderr: jest.fn()
+    };
+    testCancelationToken = {
+      isCancellationRequested: false
+    };
+    unsubscribeSpy = jest.fn();
+    subscribeSpy = jest.fn().mockReturnValue({
+      unsubscribe: unsubscribeSpy
+    });
+    fromEventSpy = jest.spyOn(Observable, 'fromEvent').mockReturnValue({
+      subscribe: subscribeSpy
+    } as any);
+    intervalSpy = jest.spyOn(Observable, 'interval').mockReturnValue({
+      subscribe: subscribeSpy
+    } as any);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
```

Comment:
I've worked on projects previously where we set this in the global after for all tests.  I'm not sure if that's the right move or not but we could put that in place.

Reply:
That would be cool. Is there ever a situation where one would want to not have this happen after a test?

Reply:
No I wouldn't think so. It wouldn't be goodto have one test depending on setup from another so I think we could just reset all Mocks by default.

Reply:
Yep sounds good to me!